### PR TITLE
feat(issue-alerts): detect incompatible rules

### PR DESCRIPTION
**Pull request by Zandew**
Adds a banner below incompatible rule.
Screen Shot 2022-10-28 at 10 52 28 AM

The logic behind detecting these incompatible rules is still very simple and can be expanded on later if the function gets too complicated. An idea for this is in the [doc](https://www.notion.so/sentry/Issue-Alert-Previews-Testing-ba0d8a43ea564ab9ba6c72b1814274f4#ec0bd3e95e4c4ab29285f6cebb7c5d8a) (still might be overkill)

### CODE REVIEW

File: static/app/views/alerts/rules/issue/index.tsx [hunk 1]
```diff
@@ -405,6 +411,36 @@ class IssueRuleEditor extends AsyncView<Props, State> {
...
+    let reappearance = 0;
+    for (let i = 0; i < conditions.length; i++) {
+      const id = conditions[i].id;
+      if (id.endsWith('FirstSeenEventCondition')) {
+        firstSeen = 1;
+      } else if (id.endsWith('RegressionEventCondition')) {
+        regression = 1;
+      } else if (id.endsWith('ReappearanceEventCondition')) {
+        reappearance = 1;
+      }
```
Comment (rjo100) [inline static/app/views/alerts/rules/issue/index.tsx:436]: change this to a one liner with a tuple
https://docs.python.org/3/library/stdtypes.html#str.endswith

`.endwith(('FirstSeenEC', 'RegressionEC', 'ReappearEC'))`

Comment (rjo100) [inline static/app/views/alerts/rules/issue/index.tsx:436]: can we use a shared counter here?

File: static/app/views/alerts/rules/issue/index.tsx [hunk 2]
```diff
@@ -405,6 +411,36 @@ class IssueRuleEditor extends AsyncView<Props, State> {
...
+  checkIncompatibleRule = debounce(() => {
+    const {rule} = this.state;
+    if (
+      !rule ||
+      !this.props.organization.features.includes('issue-alert-incompatible-rules')
+    ) {
+      return;
+    }
+
+    const conditions = rule.conditions;
```
Comment (rjo100) [inline static/app/views/alerts/rules/issue/index.tsx:424]: shouldn't this only evaluate `true` if we're using `all` and not `any`?

File: static/app/views/alerts/rules/issue/index.tsx [hunk 3]
```diff
@@ -405,6 +411,36 @@ class IssueRuleEditor extends AsyncView<Props, State> {
...
+    let firstSeen = 0;
+    let regression = 0;
+    let reappearance = 0;
```
Comment (scttcper) [inline static/app/views/alerts/rules/issue/index.tsx:427]: could use `conditions.some(condition => condition.id.endsWith())` instead of a for loop

Comment (scttcper) [reply static/app/views/alerts/rules/issue/index.tsx:427]: ah wait you're showing which condition is incompatible

File: static/app/views/alerts/rules/issue/index.tsx (hunk 1 above)
Comment (Zandew) [reply static/app/views/alerts/rules/issue/index.tsx:436]: technically they can have duplicates

**Review by scttcper — APPROVED** [review]
seems okay to me, i do worry we're making a lot of state changes and comparisons in componentDidUpdate since the alert forms are already really about rerendering

**Pull request merged**